### PR TITLE
connectors-ci: dockerd as a singleton service

### PR DESCRIPTION
**Pull request by alafanechere**
## What
This PR is an attempt at solving #26862 
Creating a dockerd service per connector context created a sheer amount of volumes and sidecar containers.
This PR is an attempt at using a singleton dockerd service shared amongs all connector context in the test pipeline.

## How
* Instantiate a global dockerd engine
* Assign this dockerd servide to each context on a `PipelineContext.dockerd_service`attribute
* Share /tmp with engine and all bound containers.

### CODE REVIEW

**Comment by alafanechere** [comment]
Tryout a nightly build on this branch: https://github.com/airbytehq/airbyte/actions/runs/5179187513

**Comment by alafanechere** [comment]
Trying out a nightly build on this branch: 
https://github.com/airbytehq/airbyte/actions/runs/5181568821/jobs/9337250695

https://alpha.dagger.cloud/runs/9476a643-7c7c-4b0d-acbb-24b25807aab2

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py [hunk 1]
```diff
@@ -425,14 +405,8 @@ async def load_image_to_docker_host(context: ConnectorContext, tar_file: File, i
...
     tar_name = f"{str(uuid.uuid4())}.tar"
-    docker_cli = with_docker_cli(context, docker_service_name=docker_service_name).with_mounted_file(tar_name, tar_file)
+    docker_cli = with_docker_cli(context).with_mounted_file(tar_name, tar_file)
 
-    # Remove a previously existing image with the same tag if any.
-    docker_image_rm_exit_code = await with_exit_code(
-        docker_cli.with_env_variable("CACHEBUSTER", tar_name).with_exec(["docker", "image", "rm", image_tag])
-    )
-    if docker_image_rm_exit_code == 0:
-        context.logger.info(f"Removed an existing image tagged {image_tag}")
```
Comment (alafanechere) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:435]: This operation was superfluous and led to alarming/noisy logs.

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py [hunk 2]
```diff
@@ -239,43 +239,27 @@ async def with_ci_connector_ops(context: PipelineContext) -> Container:
     return await with_installed_python_package(context, python_with_git, CI_CONNECTOR_OPS_SOURCE_PATH, exclude=["pipelines"])
 
 
-def with_dockerd_service(
-    context: ConnectorContext, shared_volume: Optional(Tuple[str, CacheVolume]) = None, docker_service_name: Optional[str] = None
-) -> Container:
-    """Create a container running dockerd, exposing its 2375 port, can be used as the docker host for docker-in-docker use cases.
```
Comment (bnchrch) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:245]: Can we keep this nice docstring?

**Review by bnchrch — APPROVED** [review]
Just want to keep a `docstring` in! But otherwise LGTM

**Review by bnchrch — APPROVED** [review]
Just want to keep a `docstring` in! But otherwise LGTM

**Review by bnchrch — APPROVED** [review]
Just want to keep a `docstring` in! But otherwise LGTM

**Pull request merged**